cas-server-integration-ehcache/src/main/java/org/jasig/cas/ticket/registry/EhCacheTicketRegistry.java

```diff
@@ -50,7 +51,7 @@
  * @author Andrew Tillinghast
  * @since 3.5
  */
-public final class EhCacheTicketRegistry extends AbstractDistributedTicketRegistry implements InitializingBean {
+public final class EhCacheTicketRegistry extends AbstractCrypticTicketRegistry implements InitializingBean {

```

Comment:
OK, that snippets makes more sense to me. There are still a few concerns:

1. The `DefaultTicketRegistry` does not need encryption support at all. If you are running a single node CAS instance, there is no need for you to encrypt ticket data. Safety is not an issue here. Your data is kept in memory only, and never goes anywhere else. We could keep encrypted data in memory, but that seems overkill to me. The default config should remain exactly the same.

2. Using the wrapper solution breaks existing overlays because XML config is required that would be missing.

3. The idea of `AbstractCrypticRegistry` is effectively the same as adding that boolean to note encryption support. If you extend that class, you automatically get that support. For example, Ehcache would return `true` for that property while JPA may not. If you extend that class, you automatically have declared support, so there is no need to do this. The disadvantage of that boolean is, it breaks backwards compatibility because it requires API changes which we cannot do for a minor release. Alternatively, we could declare an interface, a marker interface that each registry implements to note that they support encryption. (Similar to `Serializable`)  but then again, you're back to the same solution where you extend/implement something to note support for encryption, which is the same as you extending `AbstractCrypticRegistry`. So we are not gaining much there. 

4. I don't entirely follow your description of events that would produce errors. Can you elaborate further plz? We have to consider that encryption MUST be optional. We should not change the default behavior to force that. Many ticket registries automatically provide that support automatically, or the adopter may choose to run the registry in a private network that is totally secure. The onus would be on them to figure out whether they need that support. So by default, if you have deployed Ehcache, your configuration should work exactly the same as it did before. Encryption is an extra feature you can then turn on. Whether you use composition or inheritance, you do need to create a cipher. Without it, you always run that risk. It's something you need to consult the docs for to figure out how to enable it. 

5. The biggest problem I see from a design POV is that the `innerTicketRegistry` can be anything. See #3 for more info. We need to protect against what's passed to us, because if we can make it work without any additional errors that is ideal. Another problem is that it breaks existing XML configuration in existing overlays which makes upgrades difficult. If we can make it work without any additional breakage, then why not? 

6. I still don't quite understand why the wrapper solution is better. I see the cons outweighing the pros. What benefits do we get, that are not already provided?